cpp/daal/src/algorithms/k_nearest_neighbors/bf_knn_impl.i

```diff
@@ -271,19 +277,29 @@ protected:
         {
             for (size_t kk = 0; kk < k; ++kk)
             {
-                // max(0, d) to remove negative distances before Sqrt
-                kDistances[i * k + kk] = services::internal::max<cpu, FPType>(FPType(0), heaps[i][kk].distance);
-                kIndexes[i * k + kk]   = heaps[i][kk].index;
+                if (distancesInstance->getType() == PairwiseDistanceType::euclidean)
```

Comment:
Keep for minkowski as well?
I know that now there are abs() for them. But can be changed, probably, in the future.

Reply:
+ Probably, for keeping the performance - better to move this if outside the loop

Reply:
done